go/dbagent/dbagent.go

```diff
@@ -0,0 +1,360 @@
+		conCnt := m.GetInt("con")
+		if conCnt > connectionThreshold {
+			HasActiveConnections = true
+		}
+		return nil
+	})
+	if err != nil {
+		log.Errore(err)
+	}
+	return HasActiveConnections, err
+}
+
+func getMySQLDatadirPath() (MySQLDatadirPath string, err error) {
+	query := `SELECT @@datadir`
+	err = QueryData(query, sqlutils.Args(), func(m sqlutils.RowMap) error {
+		MySQLDatadirPath = m.GetString("@@datadir")
+		return nil
+	})
+	if err != nil {
+		log.Errore(err)
+	}
+	return MySQLDatadirPath, err
+}
+
+func GetMySQLInfo() (Info MySQLInfo, err error) {
+	Info.MySQLVersion, err = GetMySQLVersion()
+	Info.MySQLDatadirPath, err = getMySQLDatadirPath()
+	Info.MySQLBackupdirPath = config.Config.MySQLBackupDir
+	Info.IsSlave, err = isSlave()
+	Info.IsMaster, err = isMaster()
+	Info.IsBinlogEnabled, err = isBinlogEnabled()
+	Info.HasActiveConnections, err = hasActiveConnections()
+	return Info, err
+}
+
+func GetMySQLDatabases() (Databases []string, err error) {
+	query := `SELECT SCHEMA_NAME FROM information_schema.schemata WHERE SCHEMA_NAME NOT IN ('information_schema','mysql','performance_schema','sys');`
+	err = QueryData(query, sqlutils.Args(), func(m sqlutils.RowMap) error {
+		db := m.GetString("SCHEMA_NAME")
+		Databases = append(Databases, db)
+		return nil
+	})
+	if err != nil {
+		log.Errore(err)
+	}
+	return Databases, err
+}
+
+func GetMySQLEngines(dbname string) (Engines []string, err error) {
+	query := `SELECT engine FROM information_schema.tables where TABLE_SCHEMA = ? and table_type = 'BASE TABLE' GROUP BY engine;`
+	err = QueryData(query, sqlutils.Args(dbname), func(m sqlutils.RowMap) error {
+		engine := m.GetString("engine")
+		Engines = append(Engines, engine)
+		return nil
+	})
+	if err != nil {
+		log.Errore(err)
+	}
+	return Engines, err
+}
+
+func GetTokuDBSize(dbname string) (TokuSize int64, err error) {
+	query := `SELECT SUM(bt_size_allocated) AS tables_size FROM information_schema.TokuDB_fractal_tree_info WHERE table_schema = ?;`
+	err = QueryData(query, sqlutils.Args(dbname), func(m sqlutils.RowMap) error {
+		TokuSize = m.GetInt64("tables_size")
+		return nil
+	})
+	if err != nil {
+		log.Errore(err)
+	}
+	return TokuSize, err
+}
+
+func GetInnoDBLogSize() (InnoDBLogSize int64, err error) {
+	query := `SELECT @@innodb_log_file_size*@@innodb_log_files_in_group AS logFileSize;`
+	err = QueryData(query, sqlutils.Args(), func(m sqlutils.RowMap) error {
+		InnoDBLogSize = m.GetInt64("logFileSize")
+		return nil
+	})
+	if err != nil {
+		log.Errore(err)
+	}
+	return InnoDBLogSize, err
+}
+
+func UserExists(username string) (UserExists bool) {
+	query := `SELECT user FROM mysql.user WHERE user=?;`
+	err := QueryData(query, sqlutils.Args(username), func(m sqlutils.RowMap) error {
+		UserExists = (len(m.GetString("user")) != 0)
+		return nil
+	})
+	if err != nil {
+		log.Errore(err)
+	}
+	return UserExists
+}
+
+func HasGrant(username string, grant string) (HasGrant bool) {
```

Comment:
Incorrect. What if the grant is on database-level?

Reply:
I agree with your point that user provisioning and management shouldn't be done by orchestrator agent. There are chef\puppet for this cases. So i've removed it in b73ea71